packages/mui-base/src/NumberInput/NumberInput.test.tsx

```diff
@@ -0,0 +1,469 @@
+import * as React from 'react';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import userEvent from '@testing-library/user-event';
+import NumberInput, { numberInputClasses, NumberInputOwnerState } from '@mui/base/NumberInput';
+import { act, createMount, createRenderer, describeConformanceUnstyled } from 'test/utils';
+
+describe('<NumberInput />', () => {
+  const mount = createMount();
+  const { render } = createRenderer();
+
+  describeConformanceUnstyled(<NumberInput />, () => ({
+    inheritComponent: 'div',
+    render,
+    mount,
+    refInstanceof: window.HTMLDivElement,
+    testComponentPropWith: 'div',
+    muiName: 'MuiNumberInput',
+    slots: {
+      root: {
+        expectedClassName: numberInputClasses.root,
+      },
+      input: {
+        expectedClassName: numberInputClasses.input,
+        testWithElement: 'input',
+      },
+      incrementButton: {
+        expectedClassName: numberInputClasses.incrementButton,
+        testWithElement: 'button',
+      },
+      decrementButton: {
+        expectedClassName: numberInputClasses.decrementButton,
+        testWithElement: 'button',
+      },
+    },
+  }));
+
+  it('should be able to attach input ref passed through props', () => {
+    const inputRef = React.createRef<HTMLInputElement>();
+    const { getByTestId } = render(
+      <NumberInput slotProps={{ input: { 'data-testid': 'input', ref: inputRef } as any }} />,
+    );
+
+    expect(inputRef.current).to.deep.equal(getByTestId('input'));
+  });
+
+  it('passes ownerState to all the slots', () => {
```

Comment:
Passing `ownerState` is tested in conformance tests, and doesn't need to be included in individual components' tests. Let me know if you found something missing, though.